[css-align] Fix serialization of place-content/items/self properties

nt1m opened the pull request:
#### e095068eb230f338c178714fff1c77568ce522fe
<pre>
[css-align] Fix serialization of place-content/items/self properties
<a href="https://bugs.webkit.org/show_bug.cgi?id=271706">https://bugs.webkit.org/show_bug.cgi?id=271706</a>
<a href="https://rdar.apple.com/125415088">rdar://125415088</a>

Reviewed by Darin Adler.

Use `getCSSPropertyValuesFor2SidesShorthand` instead of `getCSSPropertyValuesForShorthandProperties` since these shorthands have 2 longhands that can be identical.

If they are identical, serialize to the shortest form.

Also remove these shorthands from `canUseShorthandForLonghand`, so they are not treated as legacy shorthands and expanded to their longhands in `cssText`.

* LayoutTests/css3/parse-place-content.html:
* LayoutTests/css3/parse-place-items.html:
* LayoutTests/css3/parse-place-self.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-align/content-distribution/place-content-shorthand-006-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-align/default-alignment/place-items-shorthand-006-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-align/parsing/place-content-computed-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-align/parsing/place-items-computed-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-align/parsing/place-self-computed-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-align/self-alignment/place-self-shorthand-006-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-grid/alignment/grid-content-alignment-with-abspos-001-expected.txt:
* Source/WebCore/css/ComputedStyleExtractor.cpp:
(WebCore::ComputedStyleExtractor::valueForPropertyInStyle const):
* Source/WebCore/css/StyleProperties.cpp:
(WebCore::canUseShorthandForLonghand):

Canonical link: <a href="https://commits.webkit.org/281476@main">https://commits.webkit.org/281476@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/1b01c758ba1f25cbdb7979d174bb16093407897c

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/60060 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/39408 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/12612 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/63978 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/10590 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/62190 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/47080 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/10811 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/48657 "Passed tests") | [✅ 🧪 wincairo-tests](https://ews-build.webkit.org/#/builders/60/builds/7391 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/62091 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/36746 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/52019 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/29499 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/33451 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/9259 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/9510 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/55368 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/9537 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/65710 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/87/builds/3990 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/62/builds/9412 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/56011 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/86/builds/4008 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/52003 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/56162 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/13310 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/88/builds/3317 "Passed tests") | | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/35221 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/36303 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/37391 "Built successfully") | | | 
| | [  ~~🛠 watch-sim~~](https://ews-build.webkit.org/#/builders/45/builds/36047 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
<!--EWS-Status-Bubble-End-->

darinadler commented on LayoutTests/css3/parse-place-self-expected.txt:
```diff
@@ -8,7 +8,7 @@ PASS Test getting the Computed Value of place-self's longhand properties when se
 PASS Test getting the Computed Value of place-self's longhand properties when setting 'start' value through CSS.
 PASS Test getting the Computed Value of place-self's longhand properties when setting 'flex-start' value through CSS.
-PASS Test getting the Computed Value of place-self's longhand properties when setting 'end' value through CSS.
+FAIL Test getting the Computed Value of place-self's longhand properties when setting 'end' value through CSS. assert_equals: placeSelf is not what is should. expected "end end" but got "end"
```
Why is this still failing?

nt1m replied:
Ah good catch, I missed that one when updating the test

nt1m committed:
[css-align] Fix serialization of place-content/items/self properties
https://bugs.webkit.org/show_bug.cgi?id=271706
rdar://125415088

Reviewed by Darin Adler.

Use `getCSSPropertyValuesFor2SidesShorthand` instead of `getCSSPropertyValuesForShorthandProperties` since these shorthands have 2 longhands that can be identical.

If they are identical, serialize to the shortest form.

Also remove these shorthands from `canUseShorthandForLonghand`, so they are not treated as legacy shorthands and expanded to their longhands in `cssText`.

* LayoutTests/css3/parse-place-content.html:
* LayoutTests/css3/parse-place-items.html:
* LayoutTests/css3/parse-place-self.html:
* LayoutTests/imported/w3c/web-platform-tests/css/css-align/content-distribution/place-content-shorthand-006-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-align/default-alignment/place-items-shorthand-006-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-align/parsing/place-content-computed-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-align/parsing/place-items-computed-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-align/parsing/place-self-computed-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-align/self-alignment/place-self-shorthand-006-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/css/css-grid/alignment/grid-content-alignment-with-abspos-001-expected.txt:
* Source/WebCore/css/ComputedStyleExtractor.cpp:
(WebCore::ComputedStyleExtractor::valueForPropertyInStyle const):
* Source/WebCore/css/StyleProperties.cpp:
(WebCore::canUseShorthandForLonghand):

Canonical link: https://commits.webkit.org/281476@main
